pkg/controller/upgrader/upgrader.go

```diff
@@ -167,6 +185,12 @@ func (u *Upgrader) performUpgrade(addressSpaceControllerDeployment *appsv1.Deplo
 		return err
 	}
 
+	// Delete api-server deployment
```

Comment:
I don't think I see the api-server service being remove.  Also do we need to document steps to remove the api-server's service account and roles/rolebindings?

Reply:
I will add deletion of the service. Deletion of service acocunts, roles etc, I can add it as post-upgrade steps in the documentation.